reference/src/validity/unions.md

```diff
@@ -1,13 +1,232 @@
 # Validity of unions
 
 **Disclaimer**: This chapter is a work-in-progress. What's contained here
-represents the consensus from issue [#73]. The statements in here are not (yet)
+represents the consensus from [various issues][union discussion]. The statements in here are not (yet)
 "guaranteed" not to change until an RFC ratifies them.
 
-## Validity of unions with zero-sized fields
+[union discussion]: https://github.com/rust-lang/unsafe-code-guidelines/blob/master/active_discussion/unions.md
 
-A union containing a zero-sized field can contain any bit pattern. An example of such
-an union is [`MaybeUninit`].
+**Note**: For ease of reading the examples, the hypothetical type `Padded<T>` is used, which behaves identically to `T` except that writing to it clobbers any subsequent padding. This could actually be accomplished using an overaligned newtype struct, but that would make the examples more verbose for no gain in clarity. Additionally, layout of union fields is put in some comments; in these comments, the notation `n*pad` means "`n` padding bytes".
 
+## Value model
+
+The possible values of unions are not defined in terms of the values of the fields, but rather, a union's possible values are lists of bytes. The [representation relation] is trivial in both directions, except for [padding bytes][padding byte] which are uninitialized in all values.
+
+<details><summary><b>Rationale</b></summary>
+
+The following examples must be supported, and therefore impose constraints on union value behaviour. The simplest solution, by far, is to treat the value representation of unions as merely being lists of bytes. While we do not discuss every possible angle here, it should be rapidly clear from just these two examples that any other example is significantly more complicated. Please trust us that every other alternatives that sounds even half-reasonable has been examined and has some critical flaw or another.
+
+</details>
+
+**We cannot require at least all fields, or even only one field, to be valid at all times.**
```

Comment:
"at least all fields"? What's the "at least" doing?

Reply:
Good question.